tendrl/node_agent/node_sync/sds_detect.py

```diff
@@ -35,112 +35,123 @@ def sync():
+                    if dc is None or dc.detected_cluster_id is None:
+                        integration_id = str(uuid.uuid4())
+                        try:
+                            NS._int.wclient.write(
+                                integration_index_key,
+                                integration_id,
+                                prevExist=False
+                            )
+                        except etcd.EtcdAlreadyExist:
+                            pass
 
-                        while True:
-                            # Wait till provisioner node assigns
-                            # integration_id for this detected_cluster_id
-                            try:
-                                time.sleep(5)
-                                integration_id = etcd_utils.read(
-                                    integration_index_key).value
-                                break
-                            except etcd.EtcdKeyNotFound:
-                                continue
+                    _ptag = "provisioner/%s" % \
+                        NS.tendrl_context.integration_id
+                    _flow = "tendrl.flows." \
+                        "ExpandClusterWithDetectedPeers"
+                    if _ptag in NS.node_context.tags:
+                        if dc.detected_cluster_id and \
+                            dc.detected_cluster_id != sds_details.get(
+                                'detected_cluster_id'):
+                            # Gluster peer list has changed
+                            integration_id = \
+                                NS.tendrl_context.integration_id
+                            etcd_utils.write(
+                                integration_index_key,
+                                integration_id
+                            )
+                            # Let other nodes sync up with
+                            # integration_id
+                            time.sleep(10)
+                            params = {
+                                'TendrlContext.integration_id':
+                                integration_id,
+                            }
+                            payload = {
+                                "tags": [_ptag],
+                                "run": _flow,
+                                "status": "new",
+                                "parameters": params,
+                                "type": "node"
+                            }
+                            _job_id = str(uuid.uuid4())
+                            _job = Job(
+                                job_id=_job_id,
+                                status="new",
+                                payload=payload
+                            ).save()
+                            while True:
+                                _job = _job.load()
+                                if _job.status in ["finished",
+                                                   "failed"]:
```

Comment:
If job fails how user come to know about the same? I feel we need some kind of event logging here in case of failure. In case of success anyway flow would log an event already.

Reply:
If a Job fails in Tendrl, there is an event created already. Did you mean we need something more than that?

Reply:
Ah thats enough I think